Tensile/SolutionSelectionLibrary.py

```diff
@@ -74,36 +123,108 @@ def analyzeSolutionSelection( problemType, problemSizeGroups):
         print(rowIdx)
         summationKeys = getSummationKeys(row)
       else:
-        if len(row) > 0:
+        if len(row) > 1:
+          solution = solutions[rowIdx - 1]
           keyBase = makeKey(row)
           idx=7
-          name = row[0]
+          #name = row[0]
           perfData = {}
           for summationKey in summationKeys:
             key = "%s_%s" % (keyBase,summationKey)
             value = float(row[idx])
             perfData[summationKey] = value
             idx+=1
-            if key not in performanceMap:
-              performanceMap[key] = (name,value)
+
+            if not solution in solutionsHash:
+              dataMap = {}
+              solutionsHash[solution] = dataMap
+
+            updateIfGT(solutionsHash[solution], summationKey, value)
+
+            if not key in performanceMap:
+              performanceMap[key] = (solution, value)
             else:
-              (name1,value1) = performanceMap[key]
-              if value > value1:
-                performanceMap[key] = (name,value)
-          solutionData[name]=perfData
+              _,valueOld = performanceMap[key]
+              if value > valueOld:
+                performanceMap[key] = (solution, value)
       rowIdx+=1
 
     dataFile.close()
 
-  validSolutionsNames = set([])
+  validSolutions = []
+  validSolutionSet = set([])
+
   for key in performanceMap:
-    (name,_) = performanceMap[key]      
-    validSolutionsNames.add(name)
+    solution, _ = performanceMap[key]
+    validSolutionSet.add(solution)
+  
+  for validSolution in validSolutionSet:
+    dataMap = solutionsHash[validSolution]
+    validSolutions.append((validSolution,dataMap))
+  return validSolutions
+
+def analyzeSolutionSelection(problemType, selectionFileNameList, numSolutionsPerGroup, solutionGroupMap, solutionsList):
+
+  performanceMap = {}
+  solutionsHash = {}
+
+  totalIndices = problemType["TotalIndices"]
+  summationIndex = totalIndices
+  numIndices = totalIndices + problemType["NumIndicesLD"]
+  problemSizeStartIdx = 1
+  totalSizeIdx = problemSizeStartIdx + numIndices
+  solutionStartIdx = totalSizeIdx + 1
+  for fileIdx in range(0, len(selectionFileNameList)):
+    solutions = solutionsList[fileIdx]
+    selectionFileName = selectionFileNameList[fileIdx]
+    numSolutions = numSolutionsPerGroup[fileIdx]
+    rowLength = solutionStartIdx + numSolutions
+    solutionBaseKeys = []
+
+    for solution in solutions:
+      baseKey = getSolutionBaseKey(solution)
+      solutionBaseKeys.append(baseKey)
+
+    selectionfFile = open(selectionFileName, "r") 
```

Comment:
The data analysis code should be encapsulated from the csv format.

Reply:
This is actually a csv format. I am not using a csv reader.

Reply:
What I mean is that the file I/O code should not be mixed into the data analysis code.

Reply:
Also, file objects should use [`with` statements](https://docs.python.org/3.7/reference/compound_stmts.html#the-with-statement) wherever possible.